File: grading/3-extra+codegen/valid/8-4-assignstmts-logical.py
# import all necessary Python libraries
from __future__ import print_function
import copy

_GOLITE__g = 0
def _GOLITE__bar(_GOLITE__a):
    global _GOLITE__g
    print(str(_GOLITE__a).lower() if type(_GOLITE__a) is bool else _GOLITE__a,  sep=' ', end='\n')
    _GOLITE__g+= 1
    return _GOLITE__g

def main():
    global _GOLITE__g
    _GOLITE__a = False
    _GOLITE__b = 0
    _GOLITE__c = 0
    _GOLITE__a, _GOLITE__b, _GOLITE__c = ((    _GOLITE__bar("lhs1")
 == 2) or (    _GOLITE__bar("rhs1")
 == 3)), _GOLITE__g,     _GOLITE__bar("call3")

    print(str(_GOLITE__a).lower() if type(_GOLITE__a) is bool else _GOLITE__a,  sep=' ', end='\n')
    print(str(_GOLITE__b).lower() if type(_GOLITE__b) is bool else _GOLITE__b,  sep=' ', end='\n')
    print(str(_GOLITE__c).lower() if type(_GOLITE__c) is bool else _GOLITE__c,  sep=' ', end='\n')
    _GOLITE__a, _GOLITE__b, _GOLITE__c = ((    _GOLITE__bar("lhs1")
 == 4) and (    _GOLITE__bar("rhs1")
 == 6)), _GOLITE__g,     _GOLITE__bar("call3")

    print(str(_GOLITE__a).lower() if type(_GOLITE__a) is bool else _GOLITE__a,  sep=' ', end='\n')
    print(str(_GOLITE__b).lower() if type(_GOLITE__b) is bool else _GOLITE__b,  sep=' ', end='\n')
    print(str(_GOLITE__c).lower() if type(_GOLITE__c) is bool else _GOLITE__c,  sep=' ', end='\n')

if __name__ == "__main__":
    main()

File: grading/3-extra+codegen/valid/8-4-assignstmts-logical.go
//~lhs1
//~call3
//~rhs1
//~true
//~3
//~2
//~lhs1
//~call3
//~rhs1
//~true
//~6
//~5

package main

var g int = 0

func bar(a string) int {
	println(a)
	g++
	return g
}

func main() {
	var a bool

	var b, c int

	a, b, c = bar("lhs1") == 2 || bar("rhs1") == 3, g, bar("call3")

	println(a)
	println(b)
	println(c)

	a, b, c = bar("lhs1") == 4 && bar("rhs1") == 6, g, bar("call3")

	println(a)
	println(b)
	println(c)
}
